smolyay/samples.py

```diff
@@ -0,0 +1,415 @@
+import abc
+import numpy
+
+
+class UnidimensionalPointSet(abc.ABC):
+    """Set of unidimensional points
+
+    A set of unique unidimensional points within a domain.
+    """
+
+    def __init__(self, domain):
+        self._points = None
+        self._domain = None
+
+        self.domain = numpy.array(domain, dtype=float)
+        self._valid_cache = False
+
+    @property
+    def domain(self):
+        """numpy.ndarray: Domain of the `points`."""
+        return self._domain
+
+    @domain.setter
+    def domain(self, value):
+        domain = numpy.array(value, dtype=float)
+        if domain.ndim != 1 or len(domain) != 2:
```

Comment:
I think you can use a tuple comparison here:
```suggestion
        if domain.shape != (2,):
```